src/core/vector/qgsvectorlayer.cpp

```diff
@@ -1018,6 +1018,12 @@ QgsRectangle QgsVectorLayer::extent() const
 {
   QGIS_PROTECT_QOBJECT_THREAD_ACCESS
 
+  // if data is 3D, redirect to 3D extend computation, and save it as 3D extent
+  if ( mDataProvider && mDataProvider->elevationProperties() && mDataProvider->elevationProperties()->containsElevationData() )
+  {
+    return extent3D().toRectangle();
+  }
```

Comment:
This change here should be revert too right, otherwise we're again potentially calculating an expensive 3d extent when we only need the 2d extent.

Reply:
my bad!